nanofi/include/api/ecu.h

```diff
@@ -0,0 +1,89 @@
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+    *
+    *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#ifndef NANOFI_INCLUDE_API_ECU_H_
+#define NANOFI_INCLUDE_API_ECU_H_
+
+#include <signal.h>
+#include "api/nanofi.h"
+#include "uthash.h"
+#include "utlist.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+typedef struct processor_params {
+    char uuid_str[37]; //key
+    struct flow_file_list * ff_list;
+    uint64_t curr_offset;
+    UT_hash_handle hh;
+} processor_params;
+
+extern processor_params * procparams;
```

Comment:
Having this is a global variable is
 - NOT thread-safe (you can't do anything that would write the hash table with anything else in parallel), meaning you have to be very careful when using ECUs in multiple threads
 - I am not sure how this would scale

These are acceptable for now, but should be documented.

Reply:
The only other way to deal with this kind of issue to not have globals is to modify the internal minifi C++ API we are using for nanofi. Since we will anyhow move away from using that, I did not want to deal with modifying that and throw away that work later. It wouldn't be a problem to have mutex protection when we deal with multiple threads.

Can you please elaborate a use case where we would have to deal with multiple threads in an ECU given we offload EFM communication to the orchestrator agent ?